enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -824,6 +825,12 @@ func (e *InProgressExecution) processOpUpdate(ctx context.Context, op *longrunni
 	if stage < e.lastStage {
 		return false, nil
 	}
+	// Log only on stage transitions or if it's been a while since we last
+	// logged.
+	if stage != e.lastStage || time.Since(e.lastLogTime) > 30*time.Second {
+		log.CtxInfof(ctx, "WaitExecution: %q in stage: %s", e.opName, stage)
```

Comment:
Can you use `log.EveryDuration(30*time.Second).CtxInfof(...)` instead of manually tracking the log-times here?

Reply:
I think that wouldn't quite be the same, because we want to log whenever the stage changes as well

Reply:
What about:
```
if stage != e.lastSage {
  log.CtxInfof(ctx, "WaitExecution: %q transitioned to stage: %s", e.opName, stage)
} else {
  log.EveryDuration(30*timeSecond).CtxInfof(ctx, "WaitExecution: %q still in stage: %s", e.opName, stage)
}
```